staging/src/k8s.io/apimachinery/pkg/apis/meta/internalversion/validation/validation.go

```diff
@@ -17,17 +17,19 @@ limitations under the License.
 package validation
 
 import (
+	"fmt"
+
 	"k8s.io/apimachinery/pkg/apis/meta/internalversion"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 )
 
 // ValidateListOptions returns all validation errors found while validating the ListOptions.
-func ValidateListOptions(options *internalversion.ListOptions) field.ErrorList {
+func ValidateListOptions(options *internalversion.ListOptions, isWatchListFeatureEnabled bool) field.ErrorList {
```

Comment:
I don't understand why you're passing this parameter explicitly.

You should just use the value of the feature gate directly below (and if you need to test it, just switch the feature gate appropriate in your tests, e.g.
```
defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.WatchList, true)()
```

Reply:
I did it to avoid importing `k8s.io/apiserver`. I think that `apimachinery` pkg should not depend on the `apiserver` pkgs

Reply:
This has been discussed in API PR already.